File: app/backend/src/frameworks_and_drivers/persistence/entities/sku_entity.rs
use sqlx::FromRow;

/// SKUエンティティ - 製品バリアント
/// Clean Architecture: Frameworks & Drivers層のデータ表現
#[derive(Debug, FromRow)]
pub struct SKUEntity {
    pub id: String,                        // TEXT型 (UUID)
    pub product_id: String,                // 外部キー
    pub sku_code: String,
    pub name: String,
    pub color_id: Option<i64>,             // INTEGER型
    pub dimensions: Option<String>,
    pub material: Option<String>,
    pub base_price: i64,                   // JPY in smallest unit
    pub sale_price: Option<i64>,           // JPY in smallest unit
    pub stock_quantity: i64,
    pub reserved_quantity: i64,
    pub low_stock_threshold: Option<i64>,
    pub display_order: i64,
    pub image_url: Option<String>,
    pub created_at: String,
    pub updated_at: String,
} 

File: app/backend-ts/src/infrastructure/database/entities/sku.entity.ts
import {
  Column,
  CreateDateColumn,
  Entity,
  JoinColumn,
  ManyToOne,
  PrimaryColumn,
  UpdateDateColumn,
} from 'typeorm';

import { ColorEntity } from './color.entity';
import { ProductEntity } from './product.entity';

@Entity('skus')
export class SkuEntity {
  @PrimaryColumn('text')
  id: string;

  @Column('text')
  product_id: string;

  @Column('text')
  sku_code: string;

  @Column('text')
  name: string;

  @Column('integer')
  color_id: number;

  @Column('text', { nullable: true })
  dimensions?: string;

  @Column('text', { nullable: true })
  material?: string;

  @Column('integer')
  base_price: number;

  @Column('integer', { nullable: true })
  sale_price?: number;

  @Column('integer', { default: 0 })
  stock_quantity: number;

  @Column('integer', { default: 0 })
  reserved_quantity: number;

  @Column('integer', { default: 5 })
  low_stock_threshold: number;

  @Column('integer', { default: 0 })
  display_order: number;

  @Column('text', { nullable: true })
  image_url?: string;

  @CreateDateColumn()
  created_at: Date;

  @UpdateDateColumn()
  updated_at: Date;

  @ManyToOne(() => ProductEntity, (product) => product.skus)
  @JoinColumn({ name: 'product_id' })
  product: ProductEntity;

  @ManyToOne(() => ColorEntity)
  @JoinColumn({ name: 'color_id' })
  color: ColorEntity;
}
